user/build-environment-updates/2017-07-12.md

```diff
@@ -0,0 +1,42 @@
+---
+title: Build Environment Update History
+layout: en
+permalink: /user/build-environment-updates/2017-07-12/
+category: build_env_updates
+date: 2017-07-12
+---
+
+## 2017-07-12
+
+This update applies to Ubuntu Trusty stacks on both sudo-enabled and
+container-based execution environments.
+
+### Schedule
+
+[2017-07-12 18:00 UTC](http://everytimezone.com/#2017-7-12,360,c8l)
+
+### Added
+
+- ensure `~travis/bin` dir exists at image build time
+
+### Changed
+
+- install `neo4j` via tarball instead of deb
+- bumped pre-installed Python versions to `3.4.6`, `3.5.3`, and `3.6.1` on applicable stacks
+- bumped pre-installed Ruby versions to `jruby-9.1.9.0`, `2.2.7`, `2.3.4`, and `2.4.1` on applicable stacks
+- install `mercurial` via pip instead of deb, including bump to version `4.2.2`
+- switch `language: generic` from `connie` stack to `sugilite` stack
+- bumped `elasticsearch` to `5.4.3`
+
+### Details
+
+- [travis-cookbooks diff](https://github.com/travis-ci/travis-cookbooks/compare/ab523d9...15a6f94)
+- [packer-templates diff](https://github.com/travis-ci/packer-templates/compare/35e58d2...78c07e1)
+
+#### stack-specific diffs
```

Comment:
Maybe we could add a paragraph on how to find on which stack a build is running? It could be a copy/paste from [this blog post](https://blog.travis-ci.com/2017-06-21-trusty-updates-2017-Q2-launch).